src/main/java/org/prebid/cache/handlers/PostCacheHandler.java

```diff
@@ -98,11 +98,16 @@ public PostCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposit
                         transfer.getPrefix(),
                         new Payload(transfer.getType(), transfer.getKey(), transfer.valueAsString()),
                         transfer.getExpiry(),
-                        currentDateProvider.get()
+                        currentDateProvider.get(),
+                        RandomUUID.isExternalUUID(transfer)
                 );
     }
 
     private void validateUUID(final PayloadWrapper payload, final SynchronousSink<PayloadWrapper> sink) {
+        if(payload.isExternalId() && !config.isAllowExternalUUID()) {
+            sink.error(new InvalidUUIDException("You cant specify UUID manually."));
```

Comment:
It's better to choose another wording like "Prebid cache host forbids specifying UUID in request"